src/test/java/integration/CollectionMethodsTest.java

```diff
@@ -209,19 +212,53 @@ public void canIterateCollection_withIterator() {
   @Test
   public void canIterateCollection_withListIterator() {
     ListIterator<SelenideElement> it = $$("[name=domain] option").listIterator(3);
-    assertTrue(it.hasNext()); 
-    assertTrue(it.hasPrevious()); 
+    assertTrue(it.hasNext());
+    assertTrue(it.hasPrevious());
     it.previous().shouldHave(text("@rusmail.ru"));
-    
-    assertTrue(it.hasPrevious()); 
+
+    assertTrue(it.hasPrevious());
     it.previous().shouldHave(text("@myrambler.ru"));
-    
-    assertTrue(it.hasPrevious()); 
+
+    assertTrue(it.hasPrevious());
     it.previous().shouldHave(text("@livemail.ru"));
-  
+
     assertFalse(it.hasPrevious());
-    
+
     it.next().shouldHave(text("@livemail.ru"));
     assertTrue(it.hasPrevious());
   }
+
+  @Test
+  public void canGetFirstNElements() {
+    ElementsCollection collection =  $$x("//select[@name='domain']/option");
+    collection.first(2).shouldHaveSize(2);
+    collection.first(10).shouldHaveSize(collection.size());
+
+    List<String> regularSublist = $$x("//select[@name='domain']/option").stream()
+            .map(SelenideElement::getText)
+            .collect(Collectors.toList()).subList(0, 2);
+
+    List<String> selenideSublist = collection.first(2).stream()
+            .map(SelenideElement::getText)
+            .collect(Collectors.toList());
+
+    Assert.assertEquals(regularSublist, selenideSublist);
+  }
+
+  @Test
+  public void canGetLastNElements() {
+    ElementsCollection collection =  $$x("//select[@name='domain']/option");
+    collection.last(2).shouldHaveSize(2);
+    collection.last(10).shouldHaveSize(collection.size());
```

Comment:
The expected result as a part of test logic should be explicit. 
I would change the corresponding line to:
collection.last(10).shouldHaveSize(10);

Reply:
The size of the collection in this case is 4. If you try to get elements that are out of array boundaries the method will just return the full collection instead of throwing an error. That's what that line is testing.

Reply:
ah, good, then you should put 4 explicitly in the assert, not collection.size()